packages/icon/src/Icon/Icon.tsx

```diff
@@ -27,16 +27,29 @@ const Icon = React.forwardRef<IconRef, IconProps>(
     },
     forwardedRef,
   ) => {
+    // Get SVG part of the icon
+    const unsafeIconMarkup = icon?.split(",")?.[1];
+
+    // Replace ionicon classes with telegraph classes
+    const iconMarkup = unsafeIconMarkup
+      ?.replace(/class='ionicon'/g, "")
+      ?.replace(/ionicon-stroke-width/g, "tgph-stroke-width")
+      ?.replace(/ionicon-fill-none/g, "tgph-fill-none");
+
+    if (!iconMarkup) {
+      return <></>;
+    }
```

Comment:
Curious of the reason that is driving the changes in this [commit](https://github.com/knocklabs/telegraph/pull/47/commits/2b93350b465ad67bbcad9a2c9614ba4a8803d7c2)? Not sure if I fully follow - we are using Ion icons, but not with `IonIcon`? Maybe that's something we want to explain/document more in the comment above?

Reply:
The icon that I was originally using from `@ionic/react` isn't compatible with RSC because it adds to the className prop internally outside of React's lifecycle, this was causing hydration issues. I realized that once digging into the `IonIcon` component that it's doing way more than we need it to. So, instead of dragging around a heavy-ish external library to display the icon we already get from the `ionicons` package being passed in as the `icon` prop, we can just handle displaying it ourselves. So this component now grabs the svg markup from the `icon` prop passed in, adds our classes to it to avoid the scenario of collision, and display it that way. Doing this also means that it's within the React Lifecycle properly and is compatible with RSC.

Reply:
Cool, thanks for explaining!